src/backends.mli

```diff
@@ -0,0 +1,83 @@
+(*
+ * Copyright (c) 2015 Magnus Skjegstad <magnus@skjegstad.com>
+ *
+ * Permission to use, copy, modify, and distribute this software for any
+ * purpose with or without fee is hereby granted, provided that the above
+ * copyright notice and this permission notice appear in all copies.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+ * WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
+ * MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+ * ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+ * WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN
+ * ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+ * OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+ *)
+
+type uuid = string
+type error = [ `Not_found | `Disconnected | `Unknown of string ]
+
+module type VM_BACKEND =
+sig
+  type t
+  type vm
+
+  val connect : string -> [ `Ok of t | `Error of error ] Lwt.t
+  (** **)
+  val lookup_vm_by_uuid : t -> uuid -> [ `Ok of vm | `Error of error ] Lwt.t
+  (** Lookup a VM by UUID *)
+  val lookup_vm_by_name : t -> string -> [ `Ok of vm | `Error of error ] Lwt.t
+  (** Lookup UUID of a VM by name *)
+
+  val get_state : t -> vm -> [ `Ok of Vm_state.t | `Error of error ] Lwt.t
+  (** Get VM state from [vm] type *)
```

Comment:
you might need to add blank lines between the statements as this will confuse `ocamldoc`.